src/NLog/Targets/EventLogTarget.cs

```diff
@@ -271,13 +271,7 @@ protected override void Write(LogEventInfo logEvent)
 
             EventLogEntryType entryType = GetEntryType(logEvent);
 
-            int eventId = 0;
-            string renderEventId = RenderLogEvent(EventId, logEvent);
```

Comment:
@snakefoot please help! I removed this optimized RenderLogEvent call, and not sure how to replace it efficiently with `Layout<int>`,  etc

(I prefer to remove the whole RenderLogEvent method here and move the logic that works with `Layout<T>`)

Reply:
RenderLogEvent skips allocation of StringBuilder for complex-Layout. In this situation performance is not that important. But if you plan to use it everywhere, then either performance-optimizations should be considered, or just put a warning-label on `Layout<T>` saying that it has a performance hit compared to `RenderLogEvent`

Reply:
Yeah, so we have to make RenderLogEvent `Layout<T>` complaint. But I think that could be another PR, agree?

Reply:
ow no, I will fix this!